src/stemmabench/algorithms/manuscript.py

```diff
@@ -0,0 +1,87 @@
+        - text (str): The text of the manuscript.
+    """
+
+    def __init__(self,
+                 parent: Union[ManuscriptBase, None] = None,
+                 label: Union[str, None] = None,
+                 children: list[ManuscriptBase] = [],
+                 edges: Union[list[float], None] = None,
+                 recursive: Union[dict[str,dict], None] = None,
+                 text: Union[str, None] = None) -> None:
+        """A class representing the Manuscripts that make up the nodes of a stemma.
+
+        ### Args:
+            - parent (ManuscriptBase, Optional): The parent Manuscript of the curent Manuscript. If set to None should be the root of the tree.
+            - label (str, Optional): The label denoting the text.
+            - children (list, Optional): A list of this Manuscripts children.
+            - edges (list, Optional): A list representing the distance between the edges. Is in the same order as the list of children. 
+            - recursive (dict, Optional): Dictionary representation of the current Manuscript and all its decendents. If different than None will buil all the children of the manuscript
+            - from the given list. Should only be used when instantiating a stemma from the root. 
+            - text (str, Optional): The contense of the text.
+
+        Raises:
+            - ValueError: If both recursive and lable are not specified.
+        """
+        if text:
+            self._text: Union[str, None] = text
+        if recursive:
+            self._parent: Union[ManuscriptBase, None] = None
+            self._children: list = []
+            self._label: Union[str, None] = list(recursive.keys())[0]
+            # End recursion if list of keys is empty
+            if list(recursive[self.label]) == []:
+                return None
+            # Else for each key value add a new Manuscript with dict contense
+            for lab in recursive[self.label].keys():
+                self._children.append(Manuscript(parent=self, recursive={lab:recursive[self._label][lab]}))  
+        elif label:
+            super().__init__(label, parent, children, edges)
+        else: 
+            raise ValueError("If recursive is not specified then lable must be specified.")
+
+    @property
+    def text(self):
+        return self._text
+        
+    def __eq__(self, value: object) -> bool:
+        """Returns True if both texts have the same content and the same label.
+
+        ### Args:
+            - value (object): The object to compare to.
+
+        ### Returns:
+            - bool: Value indicating if the calling object is equal to the value parameter.
+        """
+        if isinstance(value, Manuscript):
+            return value.text == self.text and value.label == self.label
+        return False
+
+    def dump(self, folder_path: str, edge_path: Union[str, None] = None) -> None:
```

Comment:
This should be moved to the stemma class as it is too "meta" and concerns the whole stemma

Reply:
Deleted dump methods in ManuscriptInTree and ManuscriptInTreeBase. Updated the dump method in Stemma.